src/features/posts.ts

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const getPosts = createAsyncThunk<Post[], User['id']>(
+  'posts/fetch',
+  userId => getUserPosts(userId),
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(getPosts.fulfilled, (state, action) => {
+        return {
+          ...state,
+          posts: action.payload,
+        };
```

Comment:
Currently, when `getPosts` is fulfilled, you're replacing the entire `posts` array with the new posts. This might not be what you want if you're implementing pagination or infinite scroll, where you'd want to append new posts to the end of the existing array. Consider changing your logic if that's the case.